platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -239,13 +244,100 @@ protected AbstractLongList(
     /**
      * Initializes the list from the given file channel. At the moment of the call all the class metadata
      * is already initialized from the file header.
+     *
      * @param sourceFileName the name of the file from which the list is initialized
      * @param fileChannel the file channel to read the list body from
      * @throws IOException if there was a problem reading the file
      */
-    protected abstract void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel)
+    protected void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel) throws IOException {
+        if (minValidIndex.get() < 0) {
+            // Empty list, nothing to read
+            return;
+        }
+
+        final int totalNumberOfChunks = calculateNumberOfChunks(size.get());
+        final int firstChunkWithDataIndex = toIntExact(minValidIndex.get() / numLongsPerChunk);
+        final int minValidIndexInChunk = toIntExact(minValidIndex.get() % numLongsPerChunk);
+
+        for (int chunkIndex = firstChunkWithDataIndex; chunkIndex < totalNumberOfChunks; chunkIndex++) {
+            final int startIndexInChunk = (chunkIndex == firstChunkWithDataIndex) ? minValidIndexInChunk : 0;
+            final long dataIndex = (long) chunkIndex * numLongsPerChunk;
+            final long dataStartIndex = dataIndex + startIndexInChunk;
+            final long dataEndCalculatedIndex = dataIndex + numLongsPerChunk - 1;
+            final long dataEndIndex = Math.min(dataEndCalculatedIndex, maxValidIndex.get());
+            final int endIndexInChunk = startIndexInChunk + toIntExact(dataEndIndex - dataStartIndex);
+
+            C chunk = readChunkData(fileChannel, chunkIndex, startIndexInChunk, endIndexInChunk);
+            setChunk(chunkIndex, chunk, startIndexInChunk, endIndexInChunk);
+        }
+    }
+
+    /**
+     * Reads data from the specified {@code fileChannel} and stores it into a chunk.
+     * The data is read from the specified range within the chunk.
+     * Subclasses must implement this method to read data from the provided {@code fileChannel}
+     * and store it into the chunk at {@code chunkIndex}.
+     *
+     * @param fileChannel the file channel to read from
+     * @param chunkIndex the index of the chunk to store the read data
+     * @param startIndex the starting position within the chunk
```

Comment:
For index params, it's usually helpful to explicitly state if they are inclusive or exclusive. In this case, both indexes are inclusive (although usually start index is inclusive, and end index is exclusive)